### ISSUE TITLE
[adding new mon] playbook generates new mon keyring

### ISSUE THREAD

**Issue by Scooty5** [ceph/ceph-ansible#5863]
**Bug Report**

What happened:
When adding new mon using `infrastructure-playbooks/add-mon.yml`, the playbook generates new `ceph-mon.keyring` inside `/etc/ceph` and it leads to two important problems:
1. on old monitors: `/etc/ceph/ceph.mon.keyring` and `/var/lib/ceph/mon/ceph-controller1/keyring` are different after running playbook(the one from /var/lib/ceph/mon is valid)
2. on newly deployed monitor: mon is deployed with a wrong(newly generated) keyring which is placed in both `/etc/ceph` and `/var/lib/ceph/mon/.../keyring` so it's out of quorum because of the authentication problem. After I corrected those keyrings, all started to work fine.
```
debug 2020-09-28T20:28:47.556+0200 7fe7a0165700  0 cephx: verify_authorizer could not decrypt ticket info: error: bad magic in decode_decrypt, 7088916086119594764 != 18374858748799134293
debug 2020-09-28T20:28:48.242+0200 7fe7a516f700 -1 mon.controller2@0(probing) e0 handle_auth_bad_method hmm, they didn't like 2 result (13) Permission denied
```

What you expected to happen:
Ansible playbook probably shouldn't regenerate this keyring while adding new mon.

How to reproduce it (minimal and precise):
Just add new mon running playbook `infrastructure-playbooks/add-mon.yml`

ceph-ansible log: https://easyupload.io/6g9qvo

**Environment**:
* OS (e.g. from /etc/os-release): Centos 8
* Kernel (e.g. `uname -a`): 4.18.0-193.19.1.el8_2.x86_64
* Docker version if applicable (e.g. `docker version`): 19.03.13
* Ansible version (e.g. `ansible-playbook --version`): 2.9.13
* ceph-ansible version (e.g. `git head or tag or stable branch`): origin/stable-5.0
* Ceph version (e.g. `ceph -v`): 15.2.5

**Comment by chandidas** [comment]
+1 . same issue found on both `stable-5.0` and `stable-4.0`

**Comment by itsafire** [comment]
+1 I am facing exactly the same problem. add mon playbook will get stuck at "fetch ceph initial keys". The newly created monitor is indeed running and not able to join the quorum:

from the monitor log file:
```
2020-11-06 10:25:29.426 7f7b3af590 -1 mon.miniceph5@0(probing) e0 handle_auth_bad_method hmm, they didn't like 2 result (13) Permission denied
2020-11-06 10:25:44.350 7f7abae590 -1 mon.miniceph5@0(probing) e0 handle_auth_bad_method hmm, they didn't like 2 result (13) Permission denied
```

The monitor is unable to join the quorum, because it is initialised with the wrong monitor keyring in /var/lib/ceph/mon/ceph-<monid>/keyring

Every time I am trying a deployment it gets some new random keyring. When copying the correct cluster monitor keyring to the keyring file and restarting the newly created monitor, it will happily join the quorum.

Environment:

OS (e.g. from /etc/os-release): Ubuntu Bionic
Kernel (e.g. uname -a): 4.4.182-rockchip64 aarch64
Ansible version (e.g. ansible-playbook --version): 2.9.14
ceph-ansible version (e.g. git head or tag or stable branch): origin/stable-4.0
Ceph version (e.g. ceph -v): 14.2.13

**Comment by jeevadotnet** [comment]
@guits fixed this with `mergify/bp/stable-5.0/pr-6064` 
https://github.com/ceph/ceph-ansible/pull/6070
https://github.com/ceph/ceph-ansible/pull/6069

**Comment by guits** [comment]
fixed in #6064
